### DISCUSSION TITLE
Why does GitHub require a mobile phone to use a FIDO2 key for 2FA?

### DISCUSSION THREAD

**Discussion by hackerb9** [community/community#66922] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

Dear Microsoft GitHub dev team,

So, with all the noise over 2FA (two-factor authentication) becoming mandatory, I think one thing that has been missed is that currently GitHub's users cannot add a FIDO2 key (such as a [Yubikey](https://www.yubico.com/))  without a mobile phone to receive an SMS text message.

I believe this is a bug as there appears to be no good security reason for it. GitHub cannot be any more sure that someone is indeed the authorized individual by sending an SMS to a number that is _not_ already on file. 

I note that adding a cell phone number for 2FA does not require any proof other than that the user has it in their possession. A FIDO2 key should be exactly the same.  

I expect that this bug is due to the common misconception that SMS is the gold standard for 2FA. In truth, FIDO2 keys are more secure than SMS, with the added benefit of privacy. 

Please look into this as soon as possible and remove the requirement for a mobile phone.

Thanks!

**Comment by corwinn** [comment]
Recently I had a negative experience with "Apple Id"; here is one more point of view: "github", do you want to permanently bind my phone number to a "github" account, so that after I delete said account, I would be no longer able to register another account using (MFA) the same phone number? A.k.a. would I be able to "unlink" my phone number - used for your MFA? Thank you.

**Reply by hackerb9** [reply]
Not an answer, but perhaps useful to know: As an experiment, I just created two GitHub accounts and used the same Google Voice number as 2FA for both. It worked, but as you suspected, @corwinn, once added there seems to be no way to remove a phone number, only replace it with another one.
